src/main/java/seedu/duke/models/schema/Schedule.java

```diff
@@ -289,8 +290,46 @@ public Module getModule(String moduleCode) throws InvalidObjectException {
         return modulesPlanned.getModule(moduleCode);
     }
 
-    public boolean canCompleteModule(String moduleCode) {
-        return true;
+    public void completeModule(Module module, ArrayList<String> modulePrereq) throws
+            FailPrereqException,
+            InvalidObjectException {
+        //we will slice from semester 0 to 1 semester before the target module,
+        //we will then check if the modules in the pre req array are
+        //we need to determine the semester of the module
+        String moduleCode = module.getModuleCode();
+        int targetIndex = modulesPlanned.getIndexByString(moduleCode);
+        int targetSem = 1;
+        int moduleCount = modulesPerSem[0];
+
+        while ((moduleCount - 1) < targetIndex) {
+            moduleCount += modulesPerSem[targetSem];
+            targetSem += 1;
+        }
+
+        List<String> partialModulesPlannedArray = modulesPlanned.getModuleCodes().subList(0, (moduleCount));
+        //we need to get the pre requisites of the module,
+        //for each prerequissite, we need to locate the module, check if its completed
+        //if not completed, throw error
+        if(modulePrereq == null){
+            modulePrereq = new ArrayList<>();
+        }
+        for(String currModule: modulePrereq){
+            if(partialModulesPlannedArray.contains(currModule) && !getModule(currModule).getCompletionStatus()){
+                throw new FailPrereqException(moduleCode);
+            }
+        }
```

Comment:
good explanations! personally, i think removing "we" can shorten the comments so they can understand faster!
`// get the prerequisite modules of the module, locate prerequisite modules, check if its completed`
`//if not completed, throw error`